nistats/tests/test_glm.py

```diff
@@ -51,257 +52,198 @@ def generate_fake_fmri_data(shapes, rk=3, affine=np.eye(4)):
     return mask, fmri_data, design_matrices
 
 
-def test_high_level_glm_with_paths():
-    shapes, rk = ((5, 6, 4, 20), (5, 6, 4, 19)), 3
-    with InTemporaryDirectory():
-        mask_file, fmri_files, design_files = write_fake_fmri_data(shapes, rk)
-        multi_session_model = FMRILinearModel(fmri_files, design_files,
-                                              mask_file)
-        multi_session_model.fit()
-        z_image, = multi_session_model.contrast([np.eye(rk)[1]] * 2)
-        assert_array_equal(z_image.get_affine(), load(mask_file).get_affine())
-        assert_true(z_image.get_data().std() < 3.)
-        # Delete objects attached to files to avoid WindowsError when deleting
-        # temporary directory
-        del z_image, fmri_files, multi_session_model
+def test_high_level_glm_one_session():
+    # New API
+    shapes, rk = [(5, 6, 7, 20)], 3
+    mask, fmri_data, design_matrices = generate_fake_fmri_data(shapes, rk)
+
+    single_session_model = FirstLevelGLM(mask=None).fit(
+        fmri_data[0], design_matrices[0])
+    assert_true(isinstance(single_session_model.masker_.mask_img_,
+                           Nifti1Image))
+
+    single_session_model = FirstLevelGLM(mask=mask).fit(
+        fmri_data[0], design_matrices[0])
+    z1, = single_session_model.transform(np.eye(rk)[:1])
+    assert_true(isinstance(z1, Nifti1Image))
 
 
 def test_high_level_glm_with_data():
+    # New API
     shapes, rk = ((7, 6, 5, 20), (7, 6, 5, 19)), 3
     mask, fmri_data, design_matrices = write_fake_fmri_data(shapes, rk)
 
-    # without mask
-    multi_session_model = FMRILinearModel(fmri_data, design_matrices,
-                                          mask=None)
-    multi_session_model.fit()
-    z_image, = multi_session_model.contrast([np.eye(rk)[1]] * 2)
-    assert_equal(np.sum(z_image.get_data() == 0), 0)
-
-    # compute the mask
-    multi_session_model = FMRILinearModel(fmri_data, design_matrices,
-                                          m=0, M=.01, threshold=0.)
-    multi_session_model.fit()
-    z_image, = multi_session_model.contrast([np.eye(rk)[1]] * 2)
+    multi_session_model = FirstLevelGLM(mask=None).fit(
+        fmri_data, design_matrices)
```

Comment:
fmri_data contains the name of files, but I do not see indication on the documentation of fit that it can also take paths instead of Nimg-like objects?